models/ssh_key.go

```diff
@@ -382,6 +382,7 @@ func calcFingerprint(publicKeyContent string) (string, error) {
 	} else if len(stdout) < 2 {
 		return "", errors.New("not enough output for calculating fingerprint: " + stdout)
 	}
+	os.Remove(tmpPath)
```

Comment:
You should still use defer before executing ssh-keygen since if the check failed the file is not removed otherwise.